@here/harp-mapview/lib/ElevationRangeSource.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TilingScheme } from "@here/harp-geoutils";
+import { TileKey } from "@here/harp-geoutils/lib/tiling/TileKey";
+
+export interface ElevationRange {
+    minElevation: number;
+    maxElevation: number;
+}
+
+export enum CalculationStatus {
+    // Calculated approximately. A more precise result may be available later.
+    PendingApproximate,
+    // Calculation completed. The result is final, won't improve upon retrying.
+    FinalPrecise
+}
+
+export interface ElevationRangeSource {
+    getElevationRange(tileKey: TileKey, range: ElevationRange): CalculationStatus;
```

Comment:
Hmm,
1) it's not documented that `range` is filled with result
2) IMHO, result type should be
```
ElevationRangeResult {
    minElevation: number;
    maxElevation: number;
    calculationStatus?: CalculationStatus;
}
```
Rationale: I just don't think it's good idea to return function results through two channels like here - output variable and return value.
It should be either return value or `out` variable if performance matters.